### ISSUE TITLE
Fix method name clash

### ISSUE THREAD

**Issue by peteryates** [DFE-Digital/govuk-components#63]
@misaka has kindly [pointed out](https://github.com/DFE-Digital/govuk-components/commit/34e697ca6c03f3438864ff2ab3568f80a5ca36c5#r42795939) that the attempt to make the library more DSL-like has created a clash of method names. ViewComponent [defines an accessor](https://github.com/github/view_component/blob/828714eaaeb62fbca8928ce588725d3ca2e7a2ff/lib/view_component/slotable.rb#L44-L53) that shares a name with the inner object.

I think the fix that will cause the least confusion could be to prefix the generated (wrapped) method with `add_`, so calling would look something like:

```ruby
govuk_accordion(id: 'def234') do |accordion|
  accordion.add_section(title: 'Section 1') do
    tag.p(class: 'govuk-body') do
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
    end
  end
end
```

**Comment by misaka** [comment]
Ah, I didn't see the comments about the pitfalls of relying on arity in this case.

This is a tough one. `add_` is good, although it's purpose isn't as clear when it comes to a non-collection slot ... first call makes sense, but what about subsequent calls? There's an [interesting comment in that issue](https://github.com/github/view_component/discussions/454#discussioncomment-78975) about how collections should behave more like an array. Would it make sense to have different semantics based on whether it's a collection or not?

Maybe it would make sense to let this be dealt-with upstream. 😛

**Comment by peteryates** [comment]
I think waiting for the upstream API to settle is definitely a good call. As this is causing you some problems at the moment I'm going to switch over to `#add_thing` format for the moment but leave this issue open so it can be properly addressed.

I hadn't really taken collections into account so when testing with normal slots no _singular_ method existed. :man_facepalming:

**Comment by peteryates** [comment]
As I commented in #64, I've been swayed and I think that `wrap_slot` probably isn't worth it. I'm inclined to remove it, it's a _bit nicer_ in some circumstances but it could be the tip of the iceberg of name clashes and confusion.

I still think that the auto-generated helpers are nice though, I'd be reluctant to remove them, but all that extra complexity to change from `accordion.slot(:item, **stuff)` to `accordion.item(**stuff)` no longer feels worth it! Thoughts welcome.

**Comment by misaka** [comment]
Ya, I agree. I was going to say the same thing last night but decided to sleep on it ... but I've not been swayed. Let's wait and see what happens upstream, I completely agree it _would_ be nicer to tidy it up.

**Comment by peteryates** [comment]
This was fixed (temporarily, at least) by #64
